src/app/server/Server.cpp

```diff
@@ -314,15 +319,41 @@ static CHIP_ERROR OpenPairingWindowUsingVerifier(uint16_t discriminator, PASEVer
     params.SetPASEVerifier(verifier);
 #endif // CONFIG_NETWORK_LAYER_BLE
 
+    if (resetAdmins == ResetAdmins::kYes)
+    {
+        gNextAvailableAdminId = 0;
+        gAdminPairings.Reset();
+    }
+
+    AdminId admin                = gNextAvailableAdminId;
+    AdminPairingInfo * adminInfo = gAdminPairings.AssignAdminId(admin);
+    VerifyOrReturnError(adminInfo != nullptr, CHIP_ERROR_NO_MEMORY);
+    gNextAvailableAdminId++;
+
+    return gRendezvousServer.WaitForPairing(std::move(params), &gTransports, &gSessions, adminInfo);
+}
+
+static CHIP_ERROR OpenCertificatePairingWindow(ResetAdmins resetAdmins, OperationalCredentialSet * opCredSet)
```

Comment:
I am not 100% clear on what `OpenCertificatePairingWindow` is conveying. Does it mean the CASE based pairing window can be closed at some point? Once, the operational credentials are provisioned on the device, it should always be available for a CASE based session (as long as the limit of active sessions has not reached).

For PASE, pairing window makes sense, as the device is not available for provisioning all the time. Once the first pairing is complete, the first fabric admin has to open the pairing window to allow further PASE pairings.